src/os/source/vectorsource.js

```diff
@@ -1683,6 +1683,22 @@ os.source.Vector.prototype.getFilteredFeatures = function(opt_allTime) {
       this.timeModel.intersection(range, true);
     }
 
+    // Hide annotations if needed
+    if (this instanceof plugin.file.kml.KMLSource) {
+      for (var i = 0, n = this.getFeatures().length; i < n; i++) {
```

Comment:
This has the desired effect, but will not be performant for a few reasons:
- This is calling `Array.includes` for each feature in the source. Worst case here is `O(n^2)`, which will not perform well for layers with thousands of features. While unlikely to have thousands of _annotated_ features, we can make this faster.
- The code is in `getFilteredFeatures`, which may be called many times per animation frame because it's used by any data consumer reacting to timeline changes.

I think the best solution here is to override the functions for managing the animation overlay (what we use to quickly animate features) in `plugin.file.kml.KMLSource`. The function overrides should be:

- `createAnimationOverlay`: Called when the timeline is opened. Call the parent function then hide all annotations.
- `disposeAnimationOverlay`: Called when the timeline is closed. Call the parent function then show all annotations (if the feature isn't hidden).
- `updateAnimationOverlay`: Called on each animation frame. Hide annotations for features in the animation overlay (these are from the previous frame), call the parent function, then show annotations for features in the animation overlay (now updated for the current frame).

This will still have an outstanding issue that show/hide via the KML tree checkbox can show the annotation when not in frame, but that should be resolved separately.

Reply:
I think that I fixed all of these problems. Could you take a look for me?

Reply:
I think that I fixed all of these comments. Could you take a look for me?